### ISSUE TITLE
Add Ruby

### ISSUE THREAD

**Issue by docgecko** [remear/jewelrybox#92]
When I choose to install a version of ruby through the Add Ruby, after a few moments of starting the install, the log screen showing the install process disappears and all that remains on the screen is the top menu bar and the rest of the screen is left totally grey.

I can still use the menu bar to navigate to the other pages, but when I go back to Add Ruby, the screen still remains grey.

I'm not sure if the installation process of the version of the ruby I chose is still installing or the whole process has now failed.

Can you check this out and let me know what has happened.

Many thanks,

Daren

**Comment by docgecko** [comment]
This happens when I try to install ruby 1.9.2-p290 on Mac OS X Lion 1.7.3

**Comment by remear** [comment]
This issue is being addressed in v1.3. It's caused by different output now
being produced with newer versions of RVM. We're working on getting 1.3
done and shipped ASAP.

**Comment by docgecko** [comment]
Ah ok, thanks for letting me know...will look forward to 1.3!

**Comment by remear** [comment]
For some reason, I can no longer reproduce this issue. I'd like to get a copy of 1.3 in your hands soon for you to test if this issue still exists. Find me in IRC Freenode #rubunity when you have a chance.

**Comment by remear** [comment]
@docgecko Are you able to participate in the v1.3 beta?

**Comment by joshwayne** [comment]
I'm experiencing the same issue with identical specs as @docgecko. I'd be happy to test the beta.

**Comment by docgecko** [comment]
Yes @remear, let me know how we get started.

**Comment by remear** [comment]
@smosh What OS are you using to run JewelryBox?

**Comment by joshwayne** [comment]
Mac OS Lion 10.7.3

**Comment by remear** [comment]
Ok. Great. I'm still somewhat worried about how v1.3 will run on 10.6. I'm running out of 10.6 machines to test it on. If you know any 10.6 testers, have them message me.

**Comment by joshwayne** [comment]
I actually have a machine running 10.6.8 that I can test on.

**Comment by remear** [comment]
If all goes according to plan, I'll probably be sending out the beta this week. I have a few more things to wrap up in the Gemset Manager and 2 minor defects to fix. Stay tuned.

**Comment by remear** [comment]
@docgecko @smosh You can grab the v1.3 beta from http://unfiniti.com/assets/JewelryBox_v1.3_beta_2068.zip

Let me know if you encounter any issues.

**Comment by docgecko** [comment]
@remear - going to download and give it a go today! will send you back any news

**Comment by steveax** [comment]
Just confirming that the issue still exists on v1.3_beta_2068. New install of Lion (10.7.4 11E53) rvm 1.13.6 (master). Attempt to add 1.8.7 results in the blank window that persists until a quit/re-launch. Now, 1.8.7 isn't supported with my compiler (Xcode 4.3.2 - LLVM) but jewelrybox should probably handle the error output more gracefully.

**Comment by remear** [comment]
@steveax I'd be very grateful if you could gist the output of installing 1.8.7 via rvm on the command-line so I can see exactly what output it generated.

**Comment by steveax** [comment]
@remear as requested:

``` shell
steve$ rvm install 1.8.7
The provided compiler '/usr/bin/gcc' is LLVM based, it is not yet fully supported by ruby and gems, please read `rvm requirements`.
```

**Comment by remear** [comment]
@steveax Does this build resolve this issue? https://docs.google.com/open?id=0B7-mS4miSnyfOGg3SzJsejJpSm8

**Comment by steveax** [comment]
@remear Afraid not - same behavior. It starts to build, then goes to the blank screen which persists until a re-launch.

**Comment by atomaka** [comment]
I'm experiencing the same issue with JewelryBox Version 1.3 Build 2156 with RVM Version 1.14.2 on Mac OSX 10.7.

**Comment by Relequestual** [comment]
I am also experiencing this issue. Same version details as @atomaka =[

**Comment by remear** [comment]
Could you see if this build resolves the issue? https://docs.google.com/open?id=0B7-mS4miSnyfdjN0bjRBa09oS0k

**Comment by defaude** [comment]
I just tried that build on a MacBook Air running OS X 10.7.4 with a fresh-installed XCode 4.3.3 (of course including the command line tools) and it produced the same error when I tried to install ruby-1.9.2-p320.

**Comment by Relequestual** [comment]
Installing gcc solved the problem for me.

**Comment by seanculver** [comment]
I have the same exact issue on version 1.3 build 2156 RVM version 1.14.5 on OS X 10.7.3, updated xcode command line tools and no luck.

**Comment by remear** [comment]
@seanculver What about with build 2175, https://docs.google.com/open?id=0B7-mS4miSnyfdjN0bjRBa09oS0k ?

**Comment by philipdbrown** [comment]
Sorry, I just tried the latest build and still having the same issue.

**Comment by remear** [comment]
Would be really helpful if everyone could post the output they receive from installing the same ruby from the command-line.

**Comment by seanculver** [comment]
I've fixed this by uninstalling xcode and installing https://github.com/kennethreitz/osx-gcc-installer

**Comment by Relequestual** [comment]
@remear The command line output from mine said I needed to install gcc, so I did, and that solved it. In the JewelryBox command line output view, it crashed before it got to that bit. I assume there is no check for the required components and it's just assumed.

**Comment by remear** [comment]
The check I added into build 2175 looks for "LLVM based, it is not yet fully supported by ruby and gems" in the output. Per all my testing, it shows the correct error view with a message instead of a blank view. If anyone has time to work directly with me in a more live setting on a debug build, we could probably resolve this issue quickly. Right now I have to shoot in the dark because I have no machines available to test against an environment with no GCC available.

**To those coming here for an immediate solution...**

This is NOT a JewelryBox issue. The only issue with JewelryBox here is it's inability to accurately detect how it's exiting when no GCC compiler is found. This issue is caused by the new output RVM displays when it can't find a GCC compiler. JewelryBox won't be responsible for installing GCC and the like for you, but it is responsible for accurately reporting back to you why installations fail. That's what we're focusing on resolving.

As it states in RVM Requirements, which you should ALWAYS read post-installing and post-updateing RVM, Ruby is not fully compatible with LLVM compilers. Yes, you "can" get it to compile more reliably with 1.9.3-p125 or newer, but some gems don't support it and you might get some Ruby segfaults. The correct way to resolve this issue is to install a GCC compiler onto your system.

All Xcode versions up to about 4.1 shipped with the standard GCC-based compiler. Starting with Xcode 4.2.x, Apple introduced the LLVM-based compiler and started to phase out the GCC-based one. Xcode 4.3 comes with only the LLVM compiler, no GCC. Ruby 1.8.x require GCC for compilation. Ruby 1.9.3-p0 has slight support for LLVM but you're lucky to get it installed. Ruby 1.9.3-p125 or newer has improved compatibility with the LLVM compiler.

So here's the breakdown of what you need:

### Ruby 1.8.x, Ruby 1.9.3-p0

Xcode 4.1 or osx-gcc-installer

### Ruby 1.9.3-p125 or newer

Xcode 4.1 or osx-gcc-installer
or
Xcode 4.2.x or newer with command-line tools installed

**Comment by Elendir** [comment]
Having Lion and XCode 4.3 I found this answer on stackoverflow:
rvm install 1.8.7 --with-gcc=clang

http://stackoverflow.com/questions/8032824/cant-install-ruby-under-lion-with-rvm-gcc-issues

Edit: don't do this for ruby older than 1.9.3-p125, see #126 for more details

**Comment by michaelscappa** [comment]
I fixed this on my system (Mountain Lion 10.8.1) by first installing OSX gcc installer: https://github.com/kennethreitz/osx-gcc-installer/

The 10.7 package will work fine.
